src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -889,6 +931,12 @@ static PyMemberDef TopicPartition_members[] = {
           " :py:const:`OFFSET_STORED`,"
           " :py:const:`OFFSET_INVALID`\n"
         },
+        {"metadata", T_OBJECT, offsetof(TopicPartition, metadata), READONLY,
+         "attribute metadata: Optional application metadata committed with the"
+         "offset (bytes)"},
+        {"metadata_size", T_PYSSIZET, offsetof(TopicPartition, metadata_size), READONLY,
```

Comment:
Probably not needed, the length is part of the metadata object and people can just do len(tp.metadata)